### DISCUSSION TITLE
Question on runOnConnect

### DISCUSSION THREAD

**Discussion by neilyoung** [bluenviron/mediamtx#1332] [Questions]
Merry Christmas,

Could you please elaborate, how "runOnConnect" is supposed to work? I configured a bash script there with absolute path, but it seems to not be called if a client connects

```
runOnConnect: sh /home/ubuntu/run.sh
```

(also tried w/o sh, same result)

The script runs fine if called from command line.

**Comment by neilyoung** [comment]
More info on this:

- I see these traces when a client connects:

```
2022/12/26 13:56:06 INF [RTSP] [conn 192.168.188.53:65170] opened
2022/12/26 13:56:06 INF [RTSP] [session 961425106] created by 192.168.188.53:65170
2022/12/26 13:56:06 INF [RTSP] [session 961425106] is reading from path 'abcde', with UDP, 1 track (H264)
```

- I suppose "opened" is generated here: https://github.com/aler9/rtsp-simple-server/blob/c778c049cec179d29f97f923e373a24abc184d6b/internal/core/rtsp_conn.go#L77

- But even though my configuration contains a "runOnConnect" entry, the following trace does not appear: https://github.com/aler9/rtsp-simple-server/blob/c778c049cec179d29f97f923e373a24abc184d6b/internal/core/rtsp_conn.go#L80

EDIT:
runOnRead fires. Maybe a misunderstanding

**Reply by aler9** [reply]
Hello, i checked and runOnConnect was broken. It is fixed in #1409

**Comment by neilyoung** [comment]
Thanks. I was meanwhile helping myself with onRead (or so), which is as good as onConnect

**Reply by neilyoung** [reply]
runOnRead actually

[a comment by an automated account is not shown]
